tests/test_main.py

```diff
@@ -1022,3 +1023,72 @@ class NewModel(DerivedModel, something=2):
         something = 1
 
     assert NewModel.something == 2
+
+
+def test_init_signature(capsys):
+    class SubModel(BaseModel):
+        id: int
+
+    class Model(BaseModel):
+        a: float
+        b: int = 10
+        c: SubModel
+
+    assert SubModel.__init__ is not Model.__init__
+    sig = inspect.signature(Model.__init__)
+    assert sig != inspect.signature(SubModel.__init__)
+
+    self: inspect.Parameter = sig.parameters['__pydantic_self__']
+    assert self.annotation is self.empty
+    assert self.default is self.empty
+    assert self.kind is self.POSITIONAL_OR_KEYWORD
+
+    params = sig.parameters
+    kw_only = inspect.Parameter.KEYWORD_ONLY
+    empty = inspect.Parameter.empty
+    for name, kind, annotation, default in (
+        ('a', kw_only, int, empty),
+        ('b', kw_only, float, 10),
+        ('c', kw_only, SubModel, empty),
+    ):
+        assert name in params
+        param: inspect.Parameter = params[name]
+        assert param.kind is kind
+        assert param.default == default
+
+    assert Model.__init__.__name__ == '__init__'
```

Comment:
can we add

```py
assert help(Model.__init__) == """
...
"""
```
Or equivalent to provide an easy to read check?

Reply:
I’ve done in first commit, but because of different formatting, tests on CI failed. There were inconsistency with spaces between arguments. Probably we can fix it by comparing strings without spaces

Reply:
Or different tests for different versions of python.

Reply:
Done!